inference-engine/src/gna_plugin/gna_plugin.cpp

```diff
@@ -431,6 +431,74 @@ void GNAPlugin::UpdateInputScaleFromNetwork(InferenceEngine::CNNNetwork & networ
+    });
+
+    // probing gna_primitives
+    if (irLayerAvatar != graphCompiler.dnnComponents.components.end()) {
+        auto & desc = outputsDesc[portId];
+        auto quantized = InferenceEngine::getInjectedData<QuantizedLayerParams>(layer);
+
+        desc.ptrs.resize(gnaFlags->gna_lib_async_threads_num);
+        desc.orientation = irLayerAvatar->dnnComponent.orientation_out;
+        desc.num_bytes_per_element = irLayerAvatar->dnnComponent.num_bytes_per_output;
+        desc.scale_factor = quantized != nullptr ? quantized->_dst_quant.GetScale() : 1.0f;
+        // TODO: this need to be fixed
+        desc.num_elements = irLayerAvatar->dnnComponent.num_rows_out;
+
+        // binding ptr for first infer request - then others will be setup during relocation
+        gnamem->bind_ptr(&desc.ptrs.front(), &irLayerAvatar->dnnComponent.ptr_outputs);
+        return true;;
+    }
+
+    // probing concatInfo
+    if (LayerInfo(layer).isConcat()) {
+        auto concatConnection  = graphCompiler.concat_connection.find(layer->name);
+        if (concatConnection != graphCompiler.concat_connection.end()) {
+            auto &desc = outputsDesc[portId];
+            auto quantized = InferenceEngine::getInjectedData<QuantizedLayerParams>(layer);
+
+            desc.ptrs.resize(gnaFlags->gna_lib_async_threads_num);
+            // TODO: what is orientation for concat
+            desc.orientation = kDnnInterleavedOrientation;
+            desc.num_bytes_per_element = layer->outData.front()->getPrecision().size();
+            desc.scale_factor = quantized != nullptr ? quantized->_dst_quant.GetScale() : 1.0f;
+            desc.num_elements = concatConnection->second.reserved_size / desc.num_bytes_per_element;
+
+            // binding ptr for first infer request - then others will be setup during relocation
+            gnamem->bind_ptr(&desc.ptrs.front(), &concatConnection->second.gna_ptr);
+            return true;
+        }
+    }
+
+    // probing a constant info, for constant trivial networks support
+    if (LayerInfo(layer).isConst()) {
+        auto const_blob = layer->blobs["custom"];
+        auto constConnection  = graphCompiler.const_connections.find(layer->name);
+        if (constConnection != graphCompiler.const_connections.end()) {
+            auto &desc = outputsDesc[portId];
+            auto quantized = InferenceEngine::getInjectedData<QuantizedLayerParams>(layer);
+
+            desc.ptrs.resize(gnaFlags->gna_lib_async_threads_num);
+            desc.orientation = kDnnInterleavedOrientation;
+            desc.num_bytes_per_element = layer->outData.front()->getPrecision().size();
+            desc.scale_factor = quantized != nullptr ? quantized->_dst_quant.GetScale() : 1.0f;
+            desc.num_elements = const_blob->size();
+
+            // binding ptr for first infer request - then others will be setup during relocation
+            gnamem->bind_ptr(&desc.ptrs.front(), &constConnection->second);
```

Comment:
it seems like it is 3 times the same code, except a line where num_elements is calculated

Reply:
moved the similar code to initOutput lambda to reuse it 3 times with different arguments